### ISSUE TITLE
Setting up a new GA4 property, then granting scopes, lands back in wrong state

### ISSUE THREAD

**Issue by eclarke1** [google/site-kit-wp#3550]
## Bug Description

After having already connected Site Kit with UA and GA4 properties assigned, when I go to the Analytics settings panel and change the GA4 property (blue notice dropdown) to "Set up a new property", it naturally prompts me to grant edit scopes for Analytics, since I haven't done that yet.

Once I have granted these extra scopes and land back in the Analytics settings panel, it no longer shows up in the same state: It once again has the originally set GA4 property selected instead of still showing "Set up a new property" which is what I originally wanted to do.

This is not really blocking, but confusing. At first I thought creating a new property had just failed, but then I realized it hadn't even run yet since I first have to click it again (related to our "grant extra scopes" workflow).

GA4 store is missing a snapshot store [LINK 1](https://github.com/google/site-kit-wp/blob/4d0abbf6b029c2b2bf4b57f539702864ed9001c4/assets/js/modules/analytics-4/datastore/index.js) [LINK 2](https://github.com/google/site-kit-wp/blob/4d0abbf6b029c2b2bf4b57f539702864ed9001c4/assets/js/modules/analytics/datastore/index.js#L45)

- Choose "Set up a new property".
- Then get redirected to OAuth to grant the extra scope.
- When user comes back, what should happen is that "Set up a new property" is still selected and now they can submit, but that is not the case. It instead shows what was originally set (i.e. stored in the option) on pageload.

[Related Asana task](https://app.asana.com/0/1200316599642381/1200443654792721) (featuring a screen recording)

## Steps to reproduce

1. Reset your local SK and connect SK again.
2. Activate the Analytics module and connect an account that has both UA and GA4 properties 
3. Once, the Analytics module is connected, edit the Analytics settings and select an option to create a new GA4 property
4. When you try to save settings, you should see the permissions popup. Click the "Proceed" button to grant the requested scopes.
5. After granting permissions, you will be redirected back to the Analytics module settings form which has a GA4 property selected instead of a previously selected option to create a new GA4 property.

## Screenshots

https://user-images.githubusercontent.com/873049/121917221-96b3d880-cd3d-11eb-82a1-64f95d23a3f5.mov

## Additional Context

 - PHP Version: 
 - Plugin Version [e.g. 22]

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The selected option to create a new GA4 property should remain selected even after a user comes back to the Analytics settings after granting additional scopes.

## Implementation Brief

* Merge #3565 

### Test Coverage

* N/A

### Visual Regression Changes

* N/A

## QA Brief

* Repeat steps in the `Steps to reproduce` section and make sure that the selected option to create a new GA4 property remains selected.

## Changelog entry

* Fix a problem where GA4 property selection is not preserved after granting additional scopes.

**Comment by eugene-manuilov** [comment]
This will be fixed in #3565

**Comment by wpdarren** [comment]
### QA Update: ⚠️ 

@eugene-manuilov would love some clarification here: 

1. I get the permission popup
2. I am redirected back to Analytics settings
3. No GA4 property is created at this point it says `Set up a new property` 
4. When I click on the `confirm changes` button, a new GA4 property is created.  

Please could you confirm that this is the expected behaviour?  The reason why I ask is because the QAB says `After granting permissions, you will be redirected back to the Analytics module settings form which has a GA4 property selected instead of a previously selected option to create a new GA4 property.`

![image](https://user-images.githubusercontent.com/73545194/122224475-0e077a80-ceac-11eb-9766-4e74ded671f3.png)

**Comment by eugene-manuilov** [comment]
@wpdarren yes, this is the correct behavior. Once you are redirected back to the settings form, you should still see the "Set up a new property" option selected.

**Comment by bethanylang** [comment]
@wpdarren We're also discussing this a bit in #3586 – this may end up being a new ticket to improve the workflows that send you to the Analytics side and then back to the Settings page.

**Comment by wpdarren** [comment]
@eugene-manuilov @bethanylang thank you for the clarification! 

### QA Update: Pass ✅ 

- After granting permissions, you are redirected back to the Analytics module settings form and when I click on the `confirm changes` button, a new GA4 property is created.
